app/src/main/java/com/poklad/androidifystore/presentation/ui/screens/all_products/AllProductsViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.poklad.androidifystore.presentation.ui.screens.all_products
+
+import androidx.lifecycle.viewModelScope
+import com.poklad.androidifystore.domain.model.ProductItem
+import com.poklad.androidifystore.domain.usecases.GetAllProductsUseCase
+import com.poklad.androidifystore.presentation.ui.base.BaseViewModel
+import com.poklad.androidifystore.utils.CoroutineDispatchersProvider
+import com.poklad.androidifystore.utils.Resource
+import com.poklad.androidifystore.utils.log
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class AllProductsViewModel @Inject constructor(
+    coroutineDispatchersProvider: CoroutineDispatchersProvider,
+    private val getAllProductsUseCase: GetAllProductsUseCase,
+) : BaseViewModel(coroutineDispatchersProvider) {
+    private val _products = MutableStateFlow<Resource<List<ProductItem>>>(Resource.Loading())
+    val products = _products.asStateFlow()
+    override val coroutineExceptionHandler: CoroutineExceptionHandler
+        get() = CoroutineExceptionHandler { _, throwable ->
+            _products.value = Resource.Error(throwable)
+        }
+
+    init {
+        loadProducts()
+    }
+
+    private fun loadProducts() {
+        viewModelScope.launch {
+            getAllProductsUseCase.execute(Unit)
```

Comment:
create an extension in `UseCaseSuspend` file to call `execute()` without parameters and under the hood pass `Unit`